src/app/core/services/user/user.service.ts

```diff
@@ -107,11 +108,16 @@ export class UserService {
 
     return this.appFacade.isAppTypeREST$.pipe(
       first(),
-      concatMap(isAppTypeRest =>
+      withLatestFrom(this.appFacade.currentLocale$),
+      concatMap(([isAppTypeRest, currentLocale]) =>
         this.apiService
-          .post<void>(AppFacade.getCustomerRestResource(body.customer.isBusinessCustomer, isAppTypeRest), newCustomer, {
-            captcha: pick(body, ['captcha', 'captchaAction']),
-          })
+          .post<void>(
+            AppFacade.getCustomerRestResource(body.customer.isBusinessCustomer, isAppTypeRest),
+            body.customer.isBusinessCustomer
+              ? { ...newCustomer, user: { ...newCustomer.user, preferredLanguage: currentLocale.lang ?? 'en_US' } }
+              : { ...newCustomer, preferredLanguage: currentLocale.lang ?? 'en_US' },
+            { captcha: pick(body, ['captcha', 'captchaAction']) }
+          )
```

Comment:
Does this change reflect the idea you suggested?

Reply:
not quite the way I imagined. Nevertheless: let's be realistic: How fancy can we _actually_ become?! I guess, there's an optimum. But forget about that and be *realistic*. Just one method, but at the *right place* (THIS IS IMPORTANT!!!!!!). I'm good with that 🥰 
https://www.youtube.com/watch?v=gLCduDJVksc

Reply:
Let's give it one more try with the `newCustomer$` stream. ;)